### PR TITLE
Update Channel pt 3

### PR DESCRIPTION
**Pull request by grdowns**
Instead of uninstalling then installing the extension, install and override when prompted.

### CODE REVIEW

**Review by sean-mcmanus — APPROVED** [review]

File: Extension/src/LanguageServer/extension.ts [hunk 1]
```diff
@@ -249,27 +248,42 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
...
-            return Promise.resolve();
-        } catch (error) {
-            return Promise.reject(new Error('Failed to install VSIX'));
-        }
+
+            // Timeout the process if no response is sent back. Ensures this Promise resolves/rejects
+            const timer: NodeJS.Timer = setTimeout(() => {
+                process.kill();
+                reject(new Error('Failed to receive response from VS Code script process for installation within 1s.'));
+            }, 1000);
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: I think you should allow a few more seconds to this timer.  There's no real rush to kill the process.  How about 10-15 seconds?

File: Extension/src/LanguageServer/extension.ts [hunk 2]
```diff
@@ -249,27 +248,42 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
...
-                return stdout.toString().trim();
+                try {
+                    const stdout: Buffer = execSync('which ' + vsCodeBinName);
+                    return stdout.toString().trim();
+                } catch (error) {
+                    return undefined;
+                }
             }
         }(platformInfo);
-        if (!vsCodeScriptPath) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: Why is this removed?  There is a case above that could result in vsCodeScriptPath being undefined.

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/src/LanguageServer/extension.ts (hunk 1 above)
Comment (grdowns) [reply Extension/src/LanguageServer/extension.ts]: Sure!

File: Extension/src/LanguageServer/extension.ts (hunk 2 above)
Comment (grdowns) [reply Extension/src/LanguageServer/extension.ts]: I'll add it back in

**Review by bobbrow — APPROVED** [review]
